core/src/main/java/org/elasticsearch/index/query/IndicesQueryBuilder.java

```diff
@@ -81,4 +108,75 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public String getName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        boolean currentIndexMatchesIndices = matchesIndices(parseContext, indices);
+
+        Query chosenQuery;
+        if (currentIndexMatchesIndices) {
+            chosenQuery = innerQuery.toQuery(parseContext);
+        } else {
+            chosenQuery = noMatchQuery.toQuery(parseContext);
+        }
+        return chosenQuery;
+    }
+    
+    static boolean matchesIndices(QueryParseContext context, String... indices) {
+        // NO COMMIT: we can't match indices unless we have a cluster service?
+        // for now we just randomly return true or false to make tests in IndicesQueryBuilderTest pass
```

Comment:
we should add ClusterService and IndexNameExpressionResolver to IndexQueryParseService so they get injected. Then this method could pretty much be moved to INdexQueryParseService like this:

```
public boolean matchesIndices(String... indices) {
        final String[] concreteIndices = indexNameExpressionResolver.concreteIndices(clusterService.state(), IndicesOptions.lenientExpandOpen(), indices);
        for (String index : concreteIndices) {
            if (Regex.simpleMatch(index, this.index.name())) {
                return true;
            }
        }
        return false;
    }
```

QueryShardContext would need to expose the same methd and delegate the IndexQueryParseService